src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -31,20 +34,47 @@
   [frame flag]
   (update-in frame [:flags] (comp set conj) flag))
 
+(defn- source-path
+  "Return the relative source path for the top-level class without extension."
+  [class]
+  (-> (str/replace (str class) #"\$.*" "")
+      (str/replace "." "/")))
+
 (defn analyze-fn
   "Add namespace, fn, and var to the frame map when the source is a Clojure
   function."
   [{:keys [file type class method] :as frame}]
-  (if (= :clj type)
+  (if (or (= :clj type)
+          (= :cljc type))
     (let [[ns fn & anons] (-> (repl/demunge class)
                               (str/replace #"--\d+" "")
                               (str/split #"/"))
           fn (or fn method)] ; protocol functions are not munged
       (assoc frame
              :ns  ns
              :fn  (str/join "/" (cons fn anons))
-             :var (str ns "/" fn)))
-    frame))
+             :var (str ns "/" fn)
+             ;; Full file path
+             :full-file (or (some-> (info/info-clj 'user (symbol (str ns "/" fn)))
+                                    :file
+                                    info/file-info
+                                    :file
+                                    u/transform-value)
+                            ;; invoke/invokeStatic
+                            (when-let [[_ ext] (re-matches #".*(\.clj.?)" (:file frame))]
```

Comment:
Do you need a regexp if know this is `clj` or `cljc`?

Reply:
Nope, code branch was originally elsewhere.